frontend/src/components/eventPage/EventPage.jsx

```diff
@@ -1,48 +1,61 @@
-import React, { useState, useEffect } from "react";
-import Header from "../header/Header";
-import DisplayEventInfo from "./DisplayEventInfo";
-import DisplayEventComments from "./DisplayEventComments";
+import { useState, useEffect } from 'react'
+import Header from '../header/Header'
+import DisplayEventInfo from './DisplayEventInfo'
+import DisplayEventComments from './DisplayEventComments'
+import LoadingPage from '../loadingPage/LoadingPage'
 
 const EventPage = () => {
-  const current_link = window.location.pathname;
-  const current_eventId = current_link.split("/")[2];
-  const [event, setEvent] = useState();
-  const [comments, setComments] = useState();
-  let [changeInEvent, setchangeInEvent] = useState(true);
+    const current_link = window.location.pathname
+    const current_eventId = current_link.split('/')[2]
+    const [event, setEvent] = useState()
+    const [comments, setComments] = useState()
+    let [changeInEvent, setchangeInEvent] = useState(true)
+    const [isLoading, setIsLoading] = useState(true)
 
-  useEffect(() => {
-    if (changeInEvent) {
-      fetchEvent();
-      fetchComments();
-    }
-    setchangeInEvent(false);
-  }, [event]);
+    useEffect(() => {
+        if (changeInEvent) {
+            setIsLoading(true)
+            fetchEvent()
+            fetchComments()
+        }
+        setIsLoading(false)
+        setchangeInEvent(false)
```

Comment:
What is the purpose of `changeInEvent`? I see that it is only used once, e.g it will never be set to `true`

Reply:
In EventPage.jsx its because the first times it fetches an empty array, and the code breaks

Reply:
For reference, CodePath project 2 has the same query loading logic https://github.com/CristobalMe/flixster-final/blob/main/src/MovieList.jsx#L14-L16

Reply:
To Do:
Fix setchangeInEvent

Reply:
Done